### DISCUSSION TITLE
We need to upgrade Vue and Quasar

### DISCUSSION THREAD

**Discussion by falkoschindler** [elimintz/justpy#476] [Ideas]
As already discussed in #460, JustPy serves Quasar 1.9.14, which can not easily be upgraded to 2.x, because JustPy depends on the older Vue 2 and Quasar 2 needs Vue 3. Vue 3, however, is not compatible with Vue 2, so we would need to rewrite parts of JustPy.

I just checked the end-of-life dates:

- Vue 2: [end of 2023](https://endoflife.date/vue) (active support already ended)
- Vue 3: ?
- Vue 4: ?
- Quasar 1.x: [end of 2022](https://github.com/quasarframework/quasar/blob/dev/ROADMAP.md#support-policy-and-schedule)
- Quasar 2.x: end of March 2023

In my point of view, we definitely need a roadmap on how to migrate to more recent versions of Vue (and Quasar, and other dependencies). Therefore we need to isolate JustPy's transport layer: What does it take to remotely control a JavaScript/Vue/Quasar app from Python? This is what needs to get updated every now and then when we need to make a major upgrade. Everything else, the Python objects like Div, QDiv, their attributes, events and props, ... - that's a different part of JustPy which should be rather stable and independent.

**Comment by WolfgangFahl** [comment]
Definitely so that's why i made #479  an issue now. The startup code needs to be way cleaner to make anything of this happen. 

The https://en.wikipedia.org/wiki/Big_ball_of_mud approach needs clean up.

**Reply by NormanShelley** [reply]
#479 is completed. What are the next steps in the migration roadmap to more recent versions of Vue and Quasar?

**Comment by WolfgangFahl** [comment]
@NormanShelley  - the main step is trying out things. As https://nicegui.io/ is already using never versions we might want to catch up since https://github.com/zauberzeug/nicegui/issues/168 is closed and therefore the nicegui approach is already deviating from the justpy core and nicegui is cathing way more stars (3.6 k at this time) then the original justpy does (we haven't even hit the 1k boundary yet) so there is also a lot more development going on in that project than here.
